### PR TITLE
Start of AVX2 functions

### PR DESCRIPTION
**Pull request by jackmott**
First batch of AVX2 functions, with tests. I've tried to emulate all the style conventions, let me know if I have missed some.

### CODE REVIEW

File: src/x86/avx2.rs [hunk 1]
```diff
@@ -1,3 +1,551 @@
...
+        let a = i8x32::new(0, 1, -1, 2, 
+                           -2, 3, -3, 4, 
+                           -4, 5, -5, std::i8::MAX, 
+                           std::i8::MIN + 1, 100, -100, -32, 
+                           0, 1, -1, 2,
+                           -2, 3, -3, 4,
+                           -4, 5, -5, std::i8::MAX, 
+                           std::i8::MIN + 1, 100, -100, -32);
+        let r = avx2::_mm256_abs_epi8(a);
+        let e = i8x32::new(0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, std::i8::MAX, (std::i8::MIN + 1).abs(), 100, 100, 32, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, std::i8::MAX, (std::i8::MIN + 1).abs(), 100, 100, 32);
```
Comment (BurntSushi) [inline src/x86/avx2.rs]: Please keep lines to a max of 79 columns. Long lines are very hard to read when auto-wrapped.

Also, prefer this:

```rust
let x = foo(
    0, 1, 2, 3,
    4, 5, 6, 7,
    ...);
```

over this:

```rust
let x = foo(0, 1, 2, 3,
            4, 5, 6, 7,
            ...);
```

Comment (jackmott) [inline src/x86/avx2.rs]: got it, will make adjustments sometime today, as kids allow.

Comment (BurntSushi) [inline src/x86/avx2.rs]: Thanks. :-) And thank you for working on this! It's a huge help. There are *a lot* of intrinsics to work through. :-)

Comment (jackmott) [inline src/x86/avx2.rs]: Anything to burn less coal!

**Comment by jackmott** [comment]
updated

**Comment by BurntSushi** [comment]
Great, thank you!

**Pull request merged**